### ISSUE TITLE
[macOS] Build for Apple Silicon / Universal Binary

### ISSUE THREAD

**Issue by dia3olik** [kapitainsky/RcloneBrowser#161]
I really hope this project is still going ;-)

I think it would be great if the next RB release would be an universal build with native support of the new Apple Silicon cpus.

**Comment by kapitainsky** [comment]
It is not dead - I am actually working on this right now. Things are only painfully slow to multiple of reasons. I hope I can deliver improved things soon.

**Comment by dia3olik** [comment]
Sure, no worries he he! I was just sayin' since the performance of the Apple Silicon macs is AMAZING so it would benefit a lot from the transition and it would future proof the project too ;-)

**Comment by dia3olik** [comment]
I was wondering if you managed to prepare everything to be Apple Silicon ready for the upcoming updated release ;-)

**Comment by kapitainsky** [comment]
Why you think it is needed at this stage? I mean it works perfectly under Rosetta 2.

Yes moving forward definitely it will have to be compiled for ARM but as I see this now it is not critical. Rclone Browser does not need much computing power - so even when run with code translation behaves exactly the same.

**Comment by detatik** [comment]
> I was wondering if you managed to prepare everything to be Apple Silicon ready for the upcoming updated release ;-)

I'm still on an Intel Mac, but given an app like this, what would be the advantages of having a native M1 app over running it using Rosetta?

**Comment by dia3olik** [comment]
@kapitainsky @detatik Because Rosetta 2 is still plagued with memory leaks, you can't use it for long duration jobs, the app will use more and more ram and cause excessive swapping and consequently excessive wear of the ssd and if you have a mac with less than 20-50gb of free disk space the system will crash and restart itself.

Also apps runs MILES faster when native instead of under Rosetta 2, yes, even "smaller" apps like this one.

We're talking about LATENCY here...it's insanely snappier, you'd never imagine something like that until you use an Apple Silicon mac...for the first time ever, almost everything is instantaneous, we as human beings perceive lag a lot and it's really ten steps forward imho, once you try it there's NO coming back, believe me...then you add the fact that the system is silent with no fan, barely warm and with 15/20hrs of runtime and the capability of a MacBook Air to be faster than an Intel MacBook Pro...

Why the reluctancy to take advantage of this new incredible architecture? ;-)

**Comment by detatik** [comment]
@dia3olik Thanks for the explanation. Hopefully the dev can get this implemented sooner rather than later.

**Comment by kapitainsky** [comment]
> Why the reluctancy to take advantage of this new incredible architecture? ;-)

I am reading right now how to create universal binary for QT app.

Are you sure that all problems you listed are related to RB running under Rosetta?  not rclone itself? which now is released as macOS ARM version

**Comment by dia3olik** [comment]
Yeah once I migrated to the new rclone the situation improved, I used the arm specific version under HomwBrew since long before the official one but the problem is once Rosetta is called the system can becomes unresponsive after a while, due to the aforementioned memory leaks...so the less rosetta is used the better, also rosetta apparently have also i/o limitation in bandwidth and also power consumption is much lower with native apps...lastly...x86 under mac is gone for good so...why not jump on-board ;-)

**Comment by kapitainsky** [comment]
There is no question if but when. 100% RB should be native binary for both macOS platforms. The challange is that it required Qt framework to support it - they started working on this but seems still work in progress.

**Comment by dia3olik** [comment]
Hey @kapitainsky do you have news about the new version? Thanks!!!
